src/Microsoft.Framework.CommandLineUtils/CommandLine/AnsiConsole.cs

```diff
@@ -9,22 +9,50 @@ namespace Microsoft.Framework.Runtime.Common.CommandLine
 {
     internal class AnsiConsole
     {
-        private AnsiConsole(TextWriter writer)
+        private AnsiConsole(TextWriter writer, IRuntimeEnvironment runtimeEnv)
         {
             Writer = writer;
-            OriginalForegroundColor = Console.ForegroundColor;
+
+            var useColors = runtimeEnv.OperatingSystem == "Windows" || runtimeEnv.RuntimeType == "Mono";
+
+            if (useColors)
+            {
+                WriteLine = WriteLineParse;
+                OriginalForegroundColor = Console.ForegroundColor;
+            }
+            else
+            {
+                WriteLine = WriteLineNoParse;
+            }
         }
 
+        public Action<string> WriteLine { get; private set; }
         private int _boldRecursion;
+        private static AnsiConsole _output;
```

Comment:
Designwise, I don't think we should cache `_output` and `_error` here. These are static field and the value depends on `runtimeEnv`. It will probably work fine now if `runtimeEnv` does not change but if for whatever reason `GetOutput()` was called multiple times with different `runtimeEnv` value the returned value for the second call would be wrong because it would use runtimeEnv from the first call. I think the `GetOutput` and `GetError` should just create and return the value and the code that invokes these methods should take care of caching the returned instance if it makes sense.